Runtime/Behaviours/CinemachineDollyCart.cs

```diff
@@ -74,12 +79,33 @@ void LateUpdate()
 
         void SetCartPosition(float distanceAlongPath)
         {
+            //if (m_Path != null)
+            //{
+                //m_Position = m_Path.StandardizeUnit(distanceAlongPath, m_PositionUnits);
+                //transform.position = m_Path.EvaluatePositionAtUnit(m_Position, m_PositionUnits);
+                //transform.rotation = m_Path.EvaluateOrientationAtUnit(m_Position, m_PositionUnits);
+            //}
             if (m_Path != null)
             {
-                m_Position = m_Path.StandardizeUnit(distanceAlongPath, m_PositionUnits);
-                transform.position = m_Path.EvaluatePositionAtUnit(m_Position, m_PositionUnits);
-                transform.rotation = m_Path.EvaluateOrientationAtUnit(m_Position, m_PositionUnits);
+                m_Position = StandardizePathDistance(distanceAlongPath);
+                float3 result = m_Path.EvaluatePosition(m_Position / m_Path.CalculateLength());
+                transform.position = new Vector3(result.x, result.y, result.z);
+                //transform.rotation = m_Path.EvaluateOrientationAtUnit(m_Position, m_PositionUnits);
+            }
+        }
+
+        float StandardizePathDistance(float distance)
```

Comment:
as discussed, this should be moved to a helper class, because TrackedDolly will need it too